File: 1.Lectures/9.Compute_en_paralelo_Hybrid_MPI_OPENMP/example_21_HYBRID_MPI_OpenMP/examples_9/mpi-omp-hello-world.c

/**********************************************************************
		C-DAC Tech Workshop : hyPACK-2013
                        October 15-18, 2013

   Example 1    :    Mpi-Omp_Hello_World.c

   Objective    : MPI-OpenMP program to print "Hello World"
                  This example demonstrates the use of
                  MPI LIBRARY CALLS and OPENMP Directives

   Input        : No Input is Required 

   Output       : Each thread created by each processor prints
                  Hello World along with its threadid and procesor rank.

   Created      : August-2013

   E-mail       : hpcfte@cdac.in     

**********************************************************************/

#include <stdio.h>
#include <omp.h>
#include <mpi.h>

/* Main Program */

main(int argc, char **argv)
{
	int             Numprocs, MyRank, iam;

	/* MPI - Initialization */

	MPI_Init(&argc, &argv);
	MPI_Comm_rank(MPI_COMM_WORLD, &MyRank);
	MPI_Comm_size(MPI_COMM_WORLD, &Numprocs);

	/* OpenMP Parallel Directive */
omp_set_num_threads(4); 
#pragma omp parallel private(iam)
	{
		iam = omp_get_thread_num();
		printf("Hello World is Printed By Process %d and Threadid %d\n", MyRank, iam);
	}

	/* MPI - Termination */
	MPI_Finalize();
}

File: 1.Lectures/9.Compute_en_paralelo_Hybrid_MPI_OPENMP/example_21_HYBRID_MPI_OpenMP/examples_9/mpi-omp-hello-world.f
!
!*************************************************************************
!
!	C-DAC Tech Workshop : hyPACK-2013
!            October 15-18, 2013
!
!   Example 1 : Mpi-Omp_Hello_World.f90
!
!  Objective  : MPI-OpenMP program to print "Hello World"
!
!               This example demonstrates the use of
!	            MPI_Init
!	            MPI_Comm_rank
!	            MPI_Comm_size
!              MPI_Finalize and 
!              OpenMP PARALLEL PRIVATE directive and 
!              run-time library routine OMP_GET_THREAD_NUM
!
!  Input     : 
!
!  Output    : Each thread from each processor prints the Hello World Message
!              and Rank of the process and thread identifier. 
!
!   Created  : August-2013
!
!   E-mail   : hpcfte@cdac.in     
!
!**************************************************************************

C         SUBROUTINE OMP_SET_NUM_THREADS (NUM_THREADS)
C         INTEGER NUM_THREADS
C         END SUBROUTINE OMP_SET_NUM_THREADS

         program HelloWorld
!   USE mpi

         include "mpif.h"

          integer  MyRank, Numprocs
          integer  status(MPI_STATUS_SIZE)
         integer ThreadID, OMP_GET_THREAD_NUM

!.........MPI initialization.... 

         call MPI_INIT(ierror)
         call MPI_COMM_SIZE(MPI_COMM_WORLD, Numprocs, ierror)
         call MPI_COMM_RANK(MPI_COMM_WORLD, MyRank, ierror)

C         CALL OMP_SET_NUM_THREADS(10) 
!$OMP PARALLEL PRIVATE(ThreadID)

        ThreadID = OMP_GET_THREAD_NUM()

        print*,"Hello World From Process", MyRank,"Thread", ThreadID

!$OMP END PARALLEL

         call MPI_FINALIZE( ierror )

         stop
         end



